### PR TITLE
ceph-pr-submodules: fix submodule check.

### PR DESCRIPTION
**Pull request by dmick**
Prior code never generated any output at all.

Fixes: http://tracker.ceph.com/issues/19244
Signed-off-by: Dan Mick <dan.mick@redhat.com>

### CODE REVIEW

**Comment by dmick** [comment]
Still need to test.

File: ceph-pr-submodules/build/build [hunk 1]
```diff
@@ -4,12 +4,13 @@ set -ex
...
-# the commit as recorded in the superproject, and $toplevel is the absolute
-# path to the top-level of the superproject.
-modified_submodules=$(git submodule foreach 'cd $toplevel; git diff --ignore-submodules=dirty origin/master -- $path')
+# show diffs between $ghprbTargetBranch (where the merge is going) and
+# $ghprbActiveCommit (the tip of the branch that's merging) with '...',
+# which is equivalent to diff $(git merge-base TB AC) AC, or "show
+# diff from common ancestor of the target branch and this branch with the
+# tip of this branch".  With --submodule, also show detail of diff in submodules.
+
+modified_submodules="$(git diff --submodule=log $ghprbTargetBranch...$ghprbActiveCommit | grep ^Submodule)"
```
Comment (tchaikov) [inline ceph-pr-submodules/build/build]: @ktdreyer seems that's the way to go !

Comment (ktdreyer) [inline ceph-pr-submodules/build/build]: :+1: cool

**Comment by dmick** [comment]
and indeed there were some bugs:

* ghprbActualCommit, not Active
* ghprbTargetBranch does not include origin/
* if grep fails, that causes the command to fail, which, because of -e, causes the script to fail
* it's safer and clearer to use -n "$var" than just $var

All fixed, good result at

https://jenkins.ceph.com/job/ceph-pr-submodules/5829/console

**Comment by dmick** [comment]
@ktdreyer, if you'll do a final review, I've assigned you; thanks @tchaikov too

**Comment by dmick** [comment]
@ktdreyer, you wanna take this one?

**Comment by tchaikov** [comment]
lgtm also.

**Pull request merged**